sparse/unit_test/Test_Sparse_gauss_seidel.hpp

```diff
@@ -814,7 +829,42 @@ void test_gauss_seidel_streams_rank1(
     }
   }
 
-  for (int i = 0; i < nstreams; i++) kh_v[i].destroy_gs_handle();
+  //*** Two-stage version ****
+  for (int apply_type = 0; apply_type < apply_count; ++apply_type) {
+    for (int i = 0; i < nstreams; i++)
+      Kokkos::deep_copy(instances[i], x_vector_v[i], zero);
+    run_gauss_seidel_streams(instances, kh_tsgs_v, input_mat_v, x_vector_v,
+                             y_vector_v, symmetric, m_omega, apply_type,
+                             nstreams);
+    for (int i = 0; i < nstreams; i++) {
+      KokkosBlas::axpby(instances[i], one, solution_x_v[i], -one,
+                        x_vector_v[i]);
+      mag_t result_norm_res = KokkosBlas::nrm2(instances[i], x_vector_v[i]);
+      EXPECT_LT(result_norm_res, initial_norm_res_v[i])
+          << "on stream_idx: " << i;
+    }
+  }
+  //*** Two-stage version (classic) ****
+  for (int apply_type = 0; apply_type < apply_count; ++apply_type) {
+    for (int i = 0; i < nstreams; i++)
+      Kokkos::deep_copy(instances[i], x_vector_v[i], zero);
+    run_gauss_seidel_streams(instances, kh_tsgs_classic_v, input_mat_v,
+                             x_vector_v, y_vector_v, symmetric,
+                             omega_tsgs_classic, apply_type, nstreams);
+    for (int i = 0; i < nstreams; i++) {
+      KokkosBlas::axpby(instances[i], one, solution_x_v[i], -one,
+                        x_vector_v[i]);
+      mag_t result_norm_res = KokkosBlas::nrm2(instances[i], x_vector_v[i]);
+      EXPECT_LT(result_norm_res, initial_norm_res_v[i])
+          << "on stream_idx: " << i;
+    }
+  }
+
+  for (int i = 0; i < nstreams; i++) {
+    kh_psgs_v[i].destroy_gs_handle();
+    kh_tsgs_v[i].destroy_gs_handle();
+    kh_tsgs_classic_v[i].destroy_gs_handle();
+  }
```

Comment:
```suggestion
```
When the vectors of handles go out of scope, all the gs handles will be destroyed automatically